main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -190,15 +190,17 @@ object Inliner {
     case OccurrenceAst.Expression.JumpTo(sym, tpe, purity, loc) => LiftedAst.Expr.JumpTo(sym, tpe, purity, loc)
 
     case OccurrenceAst.Expression.Let(sym, exp1, exp2, occur, tpe, purity, loc) =>
-      /// Case 1:
-      /// If `sym` is never used (it is `Dead`)  and `exp1` is pure, so it has no side effects, then it is safe to remove `sym`
-      /// Both code size and runtime are reduced
-      if (isDeadAndPure(occur, exp1.purity)) {
-        visitExp(exp2, subst0)
-      } else if (isDeadAndNotPure(occur, exp1.purity)) {
-        /// Case 2:
-        /// If `sym` is never used (it is `Dead`) and it is safe to inline then make a Stmt.
-        LiftedAst.Expr.Stmt(visitExp(exp1, subst0), visitExp(exp2, subst0), tpe, purity, loc)
+      if (isDead(occur)) {
+        if (Purity.hasSideeffects(exp1.purity)) {
```

Comment:
Why not just isPure and flip the conditions?

Reply:
fixed